functest/build_test.go

```diff
@@ -37,52 +38,85 @@ type expectedDeploymentConfig struct {
+		// need to set testData to a new variable scoped under the for loop
+		// otherwise tests run in parallel may get the wrong testData
+		// because the for loop reassigns it before the func runs
+		tt := testData
+		// call t.Run so that we can name and report on individual tests
+		t.Run(tt.testName, func(t *testing.T) {
+
+			if tt.buildah {
+				if runtime.GOOS != "linux" {
+					t.Skip()
+				}
+			}
+			stacksList := cmdtest.GetEnvStacksList()
 
-	stacksList := cmdtest.GetEnvStacksList()
+			// split the appsodyStack env variable
+			stackRaw := strings.Split(stacksList, " ")
 
-	// split the appsodyStack env variable
-	stackRaw := strings.Split(stacksList, " ")
+			// loop through the stacks
+			for i := range stackRaw {
 
-	// loop through the stacks
-	for i := range stackRaw {
-
-		t.Log("***Testing stack: ", stackRaw[i], "***")
+				t.Log("***Testing stack: ", stackRaw[i], "***")
 
-		sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
-		defer cleanup()
+				sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
+				defer cleanup()
 
-		// first add the test repo index
-		_, err := cmdtest.AddLocalRepo(sandbox, "LocalTestRepo", filepath.Join(sandbox.TestDataPath, "index.yaml"))
-		if err != nil {
-			t.Fatal(err)
-		}
+				// first add the test repo index
+				_, err := cmdtest.AddLocalRepo(sandbox, "LocalTestRepo", filepath.Join(sandbox.TestDataPath, "index.yaml"))
+				if err != nil {
+					t.Fatal(err)
+				}
 
-		// appsody init
-		t.Log("Running appsody init...")
-		_, err = cmdtest.RunAppsody(sandbox, "init", stackRaw[i])
-		if err != nil {
-			t.Fatal(err)
-		}
+				// appsody init
+				t.Log("Running appsody init...")
+				_, err = cmdtest.RunAppsody(sandbox, "init", stackRaw[i])
+				if err != nil {
+					t.Fatal(err)
+				}
 
-		// appsody build
-		_, err = cmdtest.RunAppsody(sandbox, "build")
-		if err != nil {
-			t.Fatal("The appsody build command failed: ", err)
-		}
+				// appsody build
+				_, err = cmdtest.RunAppsody(sandbox, tt.args...)
+				if err != nil {
+					t.Fatal("The appsody build command failed: ", err)
+				}
 
-		expectedImageTag := "dev.local/" + sandbox.ProjectName
-		listOutput, listErr := cmdtest.RunDockerCmdExec([]string{"images", "-q", expectedImageTag}, t)
-		if listErr != nil {
-			t.Fatal(listErr)
-		}
-		if listOutput == "" {
-			t.Errorf("Expected appsody build to create docker image '%s' but it was not found.", expectedImageTag)
-		}
+				expectedImageTag := "dev.local/" + sandbox.ProjectName
+				expectedImageTag = strings.Replace(expectedImageTag, "_", "-", -1)
+				if tt.buildah {
+					listOutput, listErr := cmdtest.RunBuildahCmdExec([]string{"images", "-q", expectedImageTag}, t)
+					if listErr != nil {
+						t.Fatal(listErr)
+					}
+					if listOutput == "" {
+						t.Errorf("Expected appsody build to create buildah image '%s' but it was not found.", expectedImageTag)
+					}
+					//delete the image
+					deleteImage(expectedImageTag, true, t)
```

Comment:
We could pass `t.Buildah` so that the code is nicer to read and follow instead of `true/false`